### ISSUE TITLE
Use tdlib as server

### ISSUE THREAD

**Issue by azaz5959** [tdlib/td#526]
I wanted to know if it is possible to use tdlib as a server for another program?
I mean suppose that we create a server on which tdlib is running and logged in,
also this server receive request via RESTful API and for example in that request we need to fetch chat list from tdlib and return the list to our client via RESTful API.
Can you provide an example of doing that?
We got problems to manage async tdlib functions with http request.
The programming language doesn't matter.
Thanks.

**Comment by norbekaiser** [comment]
This might be possible.
[getChats](https://core.telegram.org/tdlib/docs/classtd_1_1td__api_1_1get_chats.html) wil return you an list of up to 100 chat ids, if you need more the aproach slightly differs
With these Chat ids you can get use [getChat](https://core.telegram.org/tdlib/docs/classtd_1_1td__api_1_1get_chat.html) to obtain some basic chat information about the Chat.

About The Async Nature, you could design it as a blocking Function
Basically, wait with your http response until you got all the nescesary informations.
Running Receive and Handling the Responses in Between.

Another Approach would be to load the Chats first , and then startup the http server, so it can lookup in memory to the Chats. Information could be updated in the Background.

**Comment by levlam** [comment]
@azaz5959 It can be done through TDLib as well as any other interaction with Telegram, but you need to take into account that list of chats is constantly changed. You can't just get it once, you need to receive all changes to the list to make the program reliable. RESTful HTTP API is not suitable for providing such real-time updates. 
You can use TDLib remotely, but you will also need to pass updates from TDLib to remote clients via webhooks or similar technology.

**Comment by azaz5959** [comment]
@levlam 
Thanks.
Yes I want to login to a Telegram account with TDLib and provide its chats through webhooks.
You said it is possible through TDLib , 
Is it a native functionality?
Is there any documentation about doing this with TDLib?
I'm sorry to bother with lots of questions!

**Comment by levlam** [comment]
@azaz5959 Providing updates through webhooks is not a built-in functionality of TDLib, But there are some wrappers, which allows to interact with TDLib using sockets and TDLib JSON interface like https://github.com/vysheng/tdbot. You will likely need to write your own wrapper to achieve exactly what you need.
